happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
+        # Must use the same container if cloning
+        response = registry.entry_for_class(clone_source.__class__)
+    else:
+        clone_source = None
+        # Keep Device at registry for backwards compatibility but filter
+        # it out of new devices options
+        options = os.linesep.join(
+            [k for k, _ in registry.items() if k != "Device"]
+        )
+        ctnr_prompt = (
+            'Please select a container, or press enter for generic '
+            f'Ophyd Device container: {os.linesep}{options}'
+            '{}{}Selection'.format(os.linesep, os.linesep)
+        )
+        logger.debug(ctnr_prompt)
+        response = click.prompt(ctnr_prompt, default='OphydItem')
+        if response and response not in registry:
+            logger.info(f'Invalid device container {response}')
+            return
+
+    container = registry[response]
+    logger.debug(f'Contaner selected: {container.__name__}')
+
+    # Collect values for each field
+    kwargs = {}
+    for info in container.entry_info:
+        item_value = prompt_for_entry(info, clone_source=clone_source)
+        click.echo(f'Selecting value: {item_value}')
+
+        try:
+            info.enforce_value(item_value)
+            kwargs[info.key] = item_value
+        except Exception:
+            logger.info(f'Invalid value {item_value}')
+
+    device = client.create_device(container, **kwargs)
+    device.show_info()
+    if click.confirm('Please confirm the device info is correct'):
+        logger.info('Adding device')
         device.save()
-        device.show_info()
-    elif args.cmd == 'load':
-        logger.debug('Starting load block')
-        logger.info(f'Creating shell with devices {args.device_names}')
-        devices = {}
-        names = " ".join(args.device_names)
-        names = names.split()
-        for name in names:
-            devices[name] = client.load_device(name=name)
-
-        from IPython import start_ipython  # noqa
-        start_ipython(argv=['--quick'], user_ns=devices)
-    elif args.cmd == "update":
-        # parse input
-        input_ = " ".join(args.json).strip()
-        if input_ == "-":
-            items_input = json.load(sys.stdin)
+    else:
+        logger.info('Aborting')
+
+
+@happi_cli.command()
+@click.argument('name')
+@click.argument('edits', nargs=-1, type=str)
+@click.pass_context
+def edit(ctx, name, edits):
+    """
+    Change an existing entry by applying EDITS of the form: field=value
+    to the item of name NAME.
+    """
+    # retrieve client
+    client = ctx.obj
+
+    logger.debug('Starting edit block')
+    device = client.find_device(name=name)
+    for edit in edits:
+        field, value = edit.split('=', 1)
+        # validate field
+        try:
+            field = is_valid_identifier_not_keyword(field)
+        except ValueError:
+            logger.error(f'field ({field}) is not a valid keyword')
+            raise click.Abort()
```

Comment:
What's the difference of `click.Abort` and `click.UsageError` which I've seen used elsewhere here?

Reply:
For the purposes of this PR, `click.UsageError` is caught in `click.prompt`, which will re-prompt.  `click.Abort` is a RuntimeError that signals to click to terminate execution and output 1 to stderr.